src/lib.rs

```diff
@@ -416,9 +427,15 @@ pub mod pallet {
 					supersig.members.retain(|memb| !members_to_remove.contains(memb));
 					// this will never fail, because we set provider to 1 when creating the
 					// supersig
-					for _ in 0..(old_len - supersig.members.len()) {
+					let nb_removed = old_len - supersig.members.len();
+					for _ in 0..(nb_removed) {
 						frame_system::Pallet::<T>::dec_consumers(&supersig_id);
 					}
+
+					let reserve = <BalanceOf<T>>::from(size_of::<T::AccountId>() as u32)
+						.saturating_mul((nb_removed as u32).into())
+						.saturating_mul(T::PricePerBytes::get());
+					T::Currency::unreserve(&who, reserve);
```

Comment:
Get it out of the mutate closure.
Declare an unitialized variable `nb_removed` out of the closure, set it in the closure, use it out of the closure